pydoctor/templatewriter/pages/__init__.py

```diff
@@ -30,34 +31,125 @@ def signature(function: model.Function) -> str:
     return str(function.signature)
 
 class DocGetter:
-    def get(self, ob, summary=False):
+    """L{epydoc2stan} bridge."""
+    def get(self, ob:model.Documentable, summary:bool=False) -> Tag:
         if summary:
             return epydoc2stan.format_summary(ob)
         else:
-            doc = epydoc2stan.format_docstring(ob)
-            typ = epydoc2stan.type2stan(ob)
-            if typ is None:
-                return doc
+            return epydoc2stan.format_docstring(ob)
+    def get_type(self, ob: model.Documentable) -> Optional[Tag]:
+        return epydoc2stan.type2stan(ob)
+
+class BaseElement(Element, abc.ABC):
+    """
+    Common base HTML element. 
+    """
+    def __init__(self, 
+        system:Optional[model.System]=None, 
+        template_lookup:Optional[TemplateLookup]=None, 
+        loader:Optional[ITemplateLoader]=None, ) -> None:
+        """
+        Init a new pydoctor HTML element. 
+
+        The C{loader} property is usually got from the L{TemplateLookup} 
+        object but can also be set by argument.
+
+        @raises RuntimeError: If not enought information is provided 
+            to create the template loader. i.e. missing C{template_lookup} or C{loader} argument.
+            Could also be that the L{Template.renderable} property is not a L{ITemplateLoader} provider. 
+
+        @note: C{system} and C{template_lookup} can be none in special cases, like for L{LetterElement}. 
+        """
```

Comment:
Can we simplify this? The `template_lookup` is used for acquire a loader, so maybe `__init__` should always take a loader as its argument and there could be a classmethod that takes a `template_lookup` argument, acquires the loader from that and then instantiates the class using the loader.